### ISSUE TITLE
eaf 打开 epub 文件出错

### ISSUE THREAD

**Issue by loyalpartner** [manateelazycat/emacs-application-framework#314]
```
RuntimeError: not a PDF
EAF process starting...
Session restored:  3fb2-ae25-6a7b-7c6c-4301-ec25-ddf6 app.pdf-viewer.buffer /home/lee/.emacs.d/eaf/session.json
mupdf: not a PDF
Traceback (most recent call last):
  File "/home/lee/.emacs.d/.local/straight/repos/emacs-application-framework/app/pdf-viewer/buffer.py", line 382, in paintEvent
    qpixmap = self.get_page_pixmap(index, self.scale)
  File "/home/lee/.emacs.d/.local/straight/repos/emacs-application-framework/app/pdf-viewer/buffer.py", line 310, in get_page_pixmap
    self.char_dict[index] = self.get_page_char_rect_list(index)
  File "/home/lee/.emacs.d/.local/straight/repos/emacs-application-framework/app/pdf-viewer/buffer.py", line 742, in get_page_char_rect_list
    page_rawdict = self.document[page_index].getText("rawdict")
  File "/usr/lib/python3.8/site-packages/fitz/utils.py", line 463, in getText
    tp = page.getTextPage(flags)  # TextPage with or without images
  File "/usr/lib/python3.8/site-packages/fitz/fitz.py", line 3608, in getTextPage
    self.setRotation(0)
  File "/usr/lib/python3.8/site-packages/fitz/fitz.py", line 4140, in setRotation
    return _fitz.Page_setRotation(self, rot)
RuntimeError: not a PDF
```

**Comment by MatthewZMD** [comment]
@manateelazycat EAF现在还支持epub吗？我不用epub所以还没试过

**Comment by manateelazycat** [comment]
Yes, mupdf support epub book.

Can you open same book in other epub reader?

**Comment by loyalpartner** [comment]
@manateelazycat  我用fbreader是可以打开的，前面eaf打开epub没问题，昨天升级了下 eaf, 就遇到这个问题了

**Comment by MatthewZMD** [comment]
你git checkout一下之前的commit，看一看是什么时候引入的问题

**Comment by loyalpartner** [comment]
https://github.com/pymupdf/PyMuPDF/issues/540
是pymupdf的锅

**Comment by loyalpartner** [comment]
https://github.com/manateelazycat/emacs-application-framework/blob/83a25b83c4fc831a73a16dcdb617446a09c29507/app/pdf-viewer/buffer.py#L331
@HollowMan6 这几行代码会导致，eaf 打不开 epub文件,我注释下面的代码，就ok了
```
        if self.emacs_var_dict["eaf-pdf-dark-mode"] == "follow":
            col = self.handle_color(QColor(self.emacs_var_dict["eaf-emacs-theme-background-color"]), self.inverted_mode)
            page.drawRect(page.rect, color=col, fill=col, overlay=False)
```

错误如下：
```
EAF process starting...
Session is not restored, as no data about /home/lee/Nutstore Files/我的坚果云/book/ProGit.pdf.
Session restored:  8c1d-697c-f9f5-2c1d-5b78-b63b-7951 app.pdf-viewer.buffer /home/lee/.emacs.d/eaf/session.json
Session restored:  715b-a3d0-5fe8-95ac-0e5b-920d-e997 app.pdf-viewer.buffer /home/lee/.emacs.d/eaf/session.json
Traceback (most recent call last):
  File "/home/lee/.emacs.d/.local/straight/repos/emacs-application-framework/app/pdf-viewer/buffer.py", line 404, in paintEvent
    qpixmap = self.get_page_pixmap(index, self.scale)
  File "/home/lee/.emacs.d/.local/straight/repos/emacs-application-framework/app/pdf-viewer/buffer.py", line 333, in get_page_pixmap
    page.drawRect(page.rect, color=col, fill=col, overlay=False)
  File "/home/lee/.local/lib/python3.8/site-packages/fitz/utils.py", line 1380, in drawRect
    img = page.newShape()
  File "/home/lee/.local/lib/python3.8/site-packages/fitz/__init__.py", line 70, in <lambda>
    fitz.Page.newShape = lambda x: fitz.utils.Shape(x)
  File "/home/lee/.local/lib/python3.8/site-packages/fitz/utils.py", line 2383, in __init__
    raise ValueError("not a PDF")
ValueError: not a PDF

```

**Comment by HollowMan6** [comment]
> @HollowMan6 这几行代码会导致，eaf 打不开 epub文件,我注释下面的代码，就ok了
> ```
>         if self.emacs_var_dict["eaf-pdf-dark-mode"] == "follow":
>             col = self.handle_color(QColor(self.emacs_var_dict["eaf-emacs-theme-background-color"]), self.inverted_mode)
>             page.drawRect(page.rect, color=col, fill=col, overlay=False)
> ```
> 

![image](https://user-images.githubusercontent.com/43995067/89021524-3e592f80-d353-11ea-8f9b-2a1869eaec37.png)

🆗，之前没想过用EAF pdf-viewer打开epub，drawRect只支持pdf文件，我马上发个PR修复一下 ，避免打开epub的时候执行这段语句

**Comment by HollowMan6** [comment]
已在#356 中修复，现在应该解决了

**Comment by c1-g** [comment]
I've experienced the same thing when eaf-open an epub file, here's the error message
```
EAF process starting...
mupdf: not a PDF
Traceback (most recent call last):
  File "/usr/home/i/.emacs.d/site-lisp/emacs-application-framework/core/utils.py", line 58, in on_signal_received
    self._func(obj, *args, **kwargs)
  File "/home/i/.emacs.d/site-lisp/emacs-application-framework/eaf.py", line 159, in new_buffer
    self.create_app(buffer_id, str(url), "app.{0}.buffer".format(str(app_name)), str(arguments))
  File "/home/i/.emacs.d/site-lisp/emacs-application-framework/eaf.py", line 206, in create_app
    self.create_buffer(buffer_id, url, module_path, arguments)
  File "/home/i/.emacs.d/site-lisp/emacs-application-framework/eaf.py", line 220, in create_buffer
    app_buffer = module.AppBuffer(buffer_id, url, eaf_config_dir, arguments, self.emacs_var_dict, module_path)
  File "/usr/home/i/.emacs.d/site-lisp/emacs-application-framework/app/pdf-viewer/buffer.py", line 44, in __init__
    self.add_widget(PdfViewerWidget(url, config_dir, QColor(0, 0, 0, 255), buffer_id, emacs_var_dict))
  File "/usr/home/i/.emacs.d/site-lisp/emacs-application-framework/app/pdf-viewer/buffer.py", line 554, in __init__
    self.page_width = self.document.get_page_width()
  File "/usr/home/i/.emacs.d/site-lisp/emacs-application-framework/app/pdf-viewer/buffer.py", line 328, in get_page_width
    return self.document.pageCropBox(0).width
  File "/usr/local/lib/python3.8/site-packages/fitz/__init__.py", line 161, in deprecated_function
    return fname(*args, **kw)
  File "/usr/local/lib/python3.8/site-packages/fitz/fitz.py", line 4228, in page_cropbox
    val = _fitz.Document_page_cropbox(self, pno)
RuntimeError: not a PDF
```
I suspect the fix would be like #356 to disable the pdf-specific function.

**Comment by manateelazycat** [comment]
> I've experienced the same thing when eaf-open an epub file, here's the error message
> 
> ```
> EAF process starting...
> mupdf: not a PDF
> Traceback (most recent call last):
>   File "/usr/home/i/.emacs.d/site-lisp/emacs-application-framework/core/utils.py", line 58, in on_signal_received
>     self._func(obj, *args, **kwargs)
>   File "/home/i/.emacs.d/site-lisp/emacs-application-framework/eaf.py", line 159, in new_buffer
>     self.create_app(buffer_id, str(url), "app.{0}.buffer".format(str(app_name)), str(arguments))
>   File "/home/i/.emacs.d/site-lisp/emacs-application-framework/eaf.py", line 206, in create_app
>     self.create_buffer(buffer_id, url, module_path, arguments)
>   File "/home/i/.emacs.d/site-lisp/emacs-application-framework/eaf.py", line 220, in create_buffer
>     app_buffer = module.AppBuffer(buffer_id, url, eaf_config_dir, arguments, self.emacs_var_dict, module_path)
>   File "/usr/home/i/.emacs.d/site-lisp/emacs-application-framework/app/pdf-viewer/buffer.py", line 44, in __init__
>     self.add_widget(PdfViewerWidget(url, config_dir, QColor(0, 0, 0, 255), buffer_id, emacs_var_dict))
>   File "/usr/home/i/.emacs.d/site-lisp/emacs-application-framework/app/pdf-viewer/buffer.py", line 554, in __init__
>     self.page_width = self.document.get_page_width()
>   File "/usr/home/i/.emacs.d/site-lisp/emacs-application-framework/app/pdf-viewer/buffer.py", line 328, in get_page_width
>     return self.document.pageCropBox(0).width
>   File "/usr/local/lib/python3.8/site-packages/fitz/__init__.py", line 161, in deprecated_function
>     return fname(*args, **kw)
>   File "/usr/local/lib/python3.8/site-packages/fitz/fitz.py", line 4228, in page_cropbox
>     val = _fitz.Document_page_cropbox(self, pno)
> RuntimeError: not a PDF
> ```
> 
> I suspect the fix would be like #356 to disable the pdf-specific function.

@c1-g Bug has fixed in newest version.
